### PR TITLE
feat(discover): added add to discover button to saved discover queries

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Added `Add to Discover` button to saved discover queries:

image
image

Keeping `feat/discover-add-dashboard-widget` as target branch for now until merged due to dependancy.

### CODE REVIEW

File: static/app/views/eventsV2/queryList.tsx [hunk 1]
```diff
@@ -208,6 +225,21 @@ class QueryList extends React.Component<Props> {
           )}
           renderContextMenu={() => (
             <ContextMenu>
+              <Feature
+                organization={organization}
+                features={['connect-discover-and-dashboards']}
```
Comment (shruthilayaj) [inline static/app/views/eventsV2/queryList.tsx:230]: We should check for both `dashboard-edit` and `connect-discover-and-dashboards` features here

File: static/app/views/eventsV2/queryList.tsx [hunk 2]
```diff
@@ -83,6 +85,21 @@ class QueryList extends React.Component<Props> {
     });
   };
 
+  handleAddQueryToDashboard =
+    (eventView: EventView) => (event: React.MouseEvent<Element>) => {
+      const {organization} = this.props;
+      event.preventDefault();
+      event.stopPropagation();
+      openAddDashboardWidgetModal({
```
Comment (shruthilayaj) [inline static/app/views/eventsV2/queryList.tsx:93]: Can we pass defaultTitle here?

File: static/app/views/eventsV2/queryList.tsx [hunk 3]
```diff
@@ -208,6 +225,21 @@ class QueryList extends React.Component<Props> {
           )}
           renderContextMenu={() => (
             <ContextMenu>
+              <Feature
+                organization={organization}
+                features={['connect-discover-and-dashboards']}
+              >
+                {({hasFeature}) =>
+                  hasFeature && (
+                    <MenuItem
```
Comment (shruthilayaj) [inline static/app/views/eventsV2/queryList.tsx:234]: the prebuilt queries should also have the option to add to dashboard (since you can add them from the query detail page)

File: static/app/components/modals/addDashboardWidgetModal.tsx [hunk 4]
```diff
@@ -340,10 +350,18 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
...
       onUpdateWidget,
       widget: previousWidget,
+      start,
+      end,
+      statsPeriod,
     } = this.props;
     const state = this.state;
     const errors = state.errors;
 
+    const querySelection = statsPeriod
```
Comment (shruthilayaj) [inline static/app/components/modals/addDashboardWidgetModal.tsx:360]: can we do this instead of casting global selection below?
```suggestion
    const querySelection: GlobalSelection = statsPeriod
```

Comment (shruthilayaj) [reply static/app/components/modals/addDashboardWidgetModal.tsx:360]: hmm in this case, should we also be persisting environment and projects selection to the dashboard? If it gets too complicated, I think we can just use the default selection as the first pass?

File: static/app/views/eventsV2/queryList.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply static/app/views/eventsV2/queryList.tsx]: Ahh nice catch, forgot to include some of the feedback changes from the other PR into this one! Updated!

File: static/app/views/eventsV2/queryList.tsx (hunk 2 above)
Comment (edwardgou-sentry) [reply static/app/views/eventsV2/queryList.tsx:94]: 👍

File: static/app/views/eventsV2/queryList.tsx (hunk 3 above)
Comment (edwardgou-sentry) [reply static/app/views/eventsV2/queryList.tsx:257]: Updated! 👍

File: static/app/components/modals/addDashboardWidgetModal.tsx (hunk 4 above)
Comment (edwardgou-sentry) [reply static/app/components/modals/addDashboardWidgetModal.tsx]: Updated this! Had to make a slight tweak to how `querySelection` is constructed to please typescript but functionally it works the same.

File: static/app/utils/discover/eventView.tsx [hunk 5]
```diff
@@ -413,7 +413,7 @@ class EventView {
       statsPeriod: saved.range,
     });
 
-    return new EventView({
+    const eventView = new EventView({
```
Comment (wmak) [inline static/app/utils/discover/eventView.tsx:416]: nit: is this change needed?

**Review by wmak — APPROVED** [review]
LGTM

File: static/app/utils/discover/eventView.tsx (hunk 5 above)
Comment (edwardgou-sentry) [reply static/app/utils/discover/eventView.tsx]: oops no this is not needed, left it in by accident during testing. cleaned it up!

**Review by shruthilayaj — APPROVED** [review]
